### ISSUE TITLE
Problems for modeling the thermal spectral flux computed by scuff-neq

### ISSUE THREAD

**Issue by klnguyen** [HomerReid/scuff-em#68]
Hi,

I am using scuff-neq to compute the temperature-independant spectral flux radiated flux by a single SiC sphere and transferred between two SiC spheres (center-center distance = 10 micrometer).

The SiC permeability Mu and permittivity Eps are 
Mu(w)= 1 and Eps(w) = EpsInf*(1+(wL^2-wT^2)/(wT^2-w^2-i*Gamma*w))
where 
EpsInf = 6.7, wL = 1.825e14 rad/s, wT = 1.494e14 rad/s and Gamma = 8.966e11 rad/s.
The first four figures represent the spectral flux tau_rad, radiated by a signle sphere and the relative errors. N denotes the edge number of FE mesh. The black continuous curves are analytical results shown by equation (124) in Ref. [Kruger2012]. 
[Kruger2012] M. Krüger, G. Bimonte, T. Emig, and M. Kardar. Trace formulas for non equilibrium Casimir interactions, heat radiation, and heat transfer for arbitrary objects. Physical Review B 86, 115423 (2012)

Following these figures, we can observe that :
- The numerial results converge but not to the analytical results (with the relative errors 25% in the whole frequency range)
- For a small sphere (R = 0.0125 micrometer), we obtained bad numerical results for the low frequencies. They become worse for a fine mesh

The second four figures represent the spectral flux tau_transfer, exchanged by two spheres and the relative errors. The black continuous curves are reference results given by equation (137) in Ref.[Kruger2012]. We also have the relative errors around 25% in the whole frequency range. For a small sphere (R = 0.0125 micrometer), the transfered flux computed by scuff-neq is in good consistence with the reference results, especially for low frequencies, contrary to the computation of radiated flux.

We haven't understood the reason for these phenomena yet. We have the following questions :  
- Are these phenomena related to the discretization of the curved objects or related to the BEM method? We are trying to simulate the plates to verify the relative errors.
- Are the bad numerical results of the radiated flux for a small sphere (R = 0.0125 micrometer) due to the flact that the ratio between the size of FE elements and the wavelength are too small?

If you know the reason for these phenomena, please tell me. Thank you very much.
Kind regads
![figure1](https://cloud.githubusercontent.com/assets/16517022/12075012/1fa4f2a2-b172-11e5-97af-81dd3eb40d7e.png)
![figure2](https://cloud.githubusercontent.com/assets/16517022/12075013/1fa5306e-b172-11e5-8b8a-dc075bb6c517.png)

**Comment by HomerReid** [comment]
Thanks for your interest in SCUFF-EM and for filing this thorough report.

I'm not sure what is responsible for the 25% errors you are seeing. I understand the low-frequency errors as arising from large numerical cancellations in the method used to compute the power at low frequencies: the normal Poynting vector is nonzero everywhere over the surface of the sphere, but it integrates to something very small at low frequencies (almost all of the flux on any portion of the surface is cancelled by the flux on other portions), so there is an inherent numerical difficulty at low frequencies. However, this doesn't explain the 25% errors you see across a wide frequency range.

I am just in the process of preparing an updated version of SCUFF-NEQ that implements some new techniques for computing power, force and torque, which will hopefully yield better results. I will notify you when that is ready, hopefully later this month sometime.

In the meantime, the existing version of SCUFF-NEQ already offers some options for selecting the method used to compute the power, force, and torque (PFT). I suggest re-running your calculations with these options:

`` --ForceDSI --DSIRadius 1.0 --DSIPoints 302``

This will use the "displaced-surface integral" technique for computing the PFT, with the Poynting-vector integral performed over a sphere of radius 1.0 with 302 cubature points. Try varying the radius of the integration sphere and the number of cubature points to see how that affects the accuracy. (Type ``scuff-neq --help`` to see a list of the allowed values of ``--DSIPoints.``)

Also, it would be interesting to see how the errors depend on material properties and sphere radius. The Kruger formulas are themselves approximations (assuming you truncate the sum at a finite number of spherical multipoles, i.e. L=1 or 2) and the various approximations are more accurate for certain materials and frequencies than for others (see e.g. Table 1 in the paper you cited). 

To your questions:

+ I don't think the difficulty lies with the discretization or the BEM method, but rather with the technique used to compute the power, and specifically with cancellations in the Poynting-vector integral. 
+ It is true that errors can arise when the frequency is large (wavelength is small) compared to the size of the triangles, and that should be evident if you carry your calculations out to higher frequencies, but this is not the origin of the low-frequency errors.

Note that these calculations can also be performed, using a different numerical method, by my [BUFF-EM code.](https://github.com/HomerReid/buff-em)

**Comment by klnguyen** [comment]
Thank you for your response.

I will try your suggestion to use the options --ForceDSI --DSIRadius 1.0 --DSIPoints 302. However, Figure 1 in Ref.[Ried 2015] shows that the DSIPFT method is less efficient for computing the absorption and scattering efficiencies. I don't khow if this technique yields more better results for PFT computation.
[Ried 2015] M. T. H. Homer Ried and Steven G. Johnson, Efficient Computation of Power, Force, and Torque in BEM Scattering Calculations, IEEE Transactions on antennas and propagation , Vol. 63, No. 8, 2015.

For the computation of transfert flux between two spheres with DSIPFT (center-center distance =10 micrometer), how can DSIRadius be set? I think that it should be greater than 10 such that the volume created by the bounding surface contains these two spheres. That's right?

**Comment by HomerReid** [comment]
No, you would need to set the radius to something **less* than 10 microns to ensure that you are only capturing the power, force and torque on one of the two spheres. For spheres with a 10-micron separation I would set ``--DSIRadius 5.0``.

**Comment by klnguyen** [comment]
Hi,

I've just tried to apply the DISPFT method to compute the spectral flux radiated by one SiC sphere of radius 0.1 micrometer with the options you suggested : -- PRad--ForceDSI --DSIRadius 1.0 --DSIPoints 302. 
This is my results :
# scuff-neq run on ...
# data file columns: 
# 1 transform tag
# 2 omega 
# 3 (sourceObject,destObject) 
# 4 PRad flux spectral density
DEFAULT 1.000000e-01 11 -0.00000000e+00 
DEFAULT 1.090000e-01 11 -0.00000000e+00 
DEFAULT 1.180000e-01 11 -0.00000000e+00 
DEFAULT 1.270000e-01 11 -0.00000000e+00 
Could you please tell me why I got these strange flux?

I also tried scuff-neq --help to see a list of the allowed values of --DSIPoints but I get this error :
error: unknown option --help (aborting)
It seems that I don't have the option --help. Could you please give me some allowed values of --DSIPoints? 

Thanks a lot.

**Comment by HomerReid** [comment]
My mistake---I had forgotten that for DSIPFT there is no distinction between radiated and absorbed power (they both involve the integral of the normal Poynting vector over the bounding sphere). So just ask for ``--PAbs`` instead of ``--PRad`` and interpret the results as the negative of the radiated power.

As for the command-line options, if you pass ``--help`` (or any other unsupported option) to any SCUFF code you should get a list of options printed to standard input. Is it possible that the reason you aren't seeing that output is because you have directed standard input to a file, for example by saying ``scuff-neq > Output`` or something? Then the error message would show up on your terminal (because it is printed to standard error) but the information on options would be written to the file.

**Comment by ZhihaoJia16** [comment]
Hi,
Do you solve the problem of `the relative errors 25% in the whole frequency range`? I have encountered similar problem. 
I am using `scuff-neq` to compute **spectral flux** between two Gold spheres [(Fluctuating-surface-current formulation of radiative heat transfer for arbitrary geometries)](http://journals.aps.org/prb/abstract/10.1103/PhysRevB.86.220302) (center-center distance = 0.6 micrometer for R=0.2 micrometer,  center-center distance = 3 micrometer for R=1 micrometer).
I use `--EPPFT` (or `--OPFT`) instead of `--ForceDSI --DSIRadius 1.0 --DSIPoints 302`. The numerical values computed is normalized by surface area A=4*pi*R*R. My results is about 75% of references.
![shangchuan](https://cloud.githubusercontent.com/assets/16773876/14270591/32d94a7c-fb22-11e5-8fad-5a4c7cf8f1ea.jpg)
If you know the reason,please tell me.
best
thank you in advance.
